File: packages/agent/src/api/conversation.ts
/**
 * Agent API - Conversation Management
 * 会话管理 API
 */
import { fetchApi } from '@packages/core/api/core';

// ============================================================
// Chat Completions
// ============================================================

export interface ChatCompletionRequest {
  query: string;
  kb_ids: string[];
  session_id?: string;
  stream?: boolean;
  top_k?: number;
  min_score?: number;
  enable_rerank?: boolean;
  enable_expansion?: boolean;
  enable_hybrid?: boolean;
}

export interface ChatCompletionResponse {
  answer: string;
  reasoning?: string;
  citations: Array<{ index: number; doc_name: string; chunk_id: string }>;
  hallu_score?: number;
  chunks_used?: number;
}

export const chatCompletions = async (data: ChatCompletionRequest) => {
  return fetchApi<ChatCompletionResponse>('/api/v1/chat/completions', {
    method: 'POST',
    body: JSON.stringify(data),
  });
};

// ============================================================
// Conversations
// ============================================================

export interface ConversationResponse {
  id: string;
  user_id?: string;
  title: string;
  kb_ids?: string[];
  is_active: boolean;
  is_archived: boolean;
  message_count: number;
  last_message_at?: string;
  created_at: string;
  updated_at: string;
}

export interface ConversationListResponse {
  items: ConversationResponse[];
  total: number;
}

export interface ConversationCreate {
  title?: string;
  kb_ids?: string[];
}

export interface ConversationUpdate {
  title?: string;
  kb_ids?: string[];
  is_active?: boolean;
  is_archived?: boolean;
}

export const createConversation = async (data: ConversationCreate, user_id?: string) => {
  const qs = user_id ? `?user_id=${user_id}` : '';
  return fetchApi<ConversationResponse>('/api/v1/conversations' + qs, {
    method: 'POST',
    body: JSON.stringify(data),
  });
};

export const listConversations = async (params?: { user_id?: string; limit?: number; offset?: number; include_archived?: boolean }) => {
  const qs = new URLSearchParams(params as Record<string, string>).toString();
  return fetchApi<ConversationListResponse>(`/api/v1/conversations${qs ? `?${qs}` : ''}`);
};

export const updateConversation = async (conv_id: string, data: ConversationUpdate) => {
  return fetchApi<ConversationResponse>(`/api/v1/conversations/${conv_id}`, {
    method: 'PUT',
    body: JSON.stringify(data),
  });
};

export const deleteConversation = async (conv_id: string) => {
  return fetchApi<void>(`/api/v1/conversations/${conv_id}`, { method: 'DELETE' });
};

export const searchConversations = async (query: string) => {
  return fetchApi(`/api/v1/conversations/search/${encodeURIComponent(query)}`);
};

export const getConversation = async (conv_id: string) => {
  return fetchApi(`/api/v1/conversations/${conv_id}`);
};

export const addMessageToConversation = async (
  conv_id: string,
  data: { role: string; content: string; sources?: any[]; model_used?: string; latency_ms?: number }
) => {
  return fetchApi(`/api/v1/conversations/${conv_id}/messages`, {
    method: 'POST',
    body: JSON.stringify(data),
  });
};

File: backend/packages/agent/schemas/conversation.py
"""
Conversation schemas - 对话历史请求/响应模型
"""
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


class ConversationCreate(BaseModel):
    """创建对话"""
    title: Optional[str] = Field(None, description="对话标题")
    kb_ids: Optional[List[str]] = Field(None, description="关联的知识库 ID 列表")


class ConversationUpdate(BaseModel):
    """更新对话"""
    title: Optional[str] = Field(None, description="对话标题")
    is_archived: Optional[bool] = Field(None, description="是否归档")


class MessageResponse(BaseModel):
    """消息响应"""
    id: str
    conversation_id: str
    role: str
    content: str
    sources: Optional[List[dict]] = None
    token_count: Optional[int] = None
    latency_ms: Optional[int] = None
    model_used: Optional[str] = None
    message_index: int
    created_at: datetime

    class Config:
        from_attributes = True


class ConversationResponse(BaseModel):
    """对话响应"""
    id: str
    user_id: Optional[int]
    title: str
    kb_ids: Optional[List[str]] = None
    is_active: bool
    is_archived: bool
    message_count: int
    last_message_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ConversationListResponse(BaseModel):
    """对话列表响应"""
    items: List[ConversationResponse]
    total: int


class ConversationDetailResponse(BaseModel):
    """对话详情响应"""
    conversation: ConversationResponse
    messages: List[MessageResponse]


class ConversationWithMessagesResponse(BaseModel):
    """带消息的对话响应"""
    id: str
    title: str
    kb_ids: Optional[List[str]] = None
    message_count: int
    last_message_at: Optional[datetime]
    messages: List[MessageResponse]
